developer/src/tike/child/Keyman.Developer.UI.Debug.UfrmLdmlKeyboardDebug.pas

```diff
@@ -474,39 +474,50 @@     TMemoSelectionState = record
     end;
   end;
 
+
+  // Returns the length of the context items in UTF16, treating markers as a
+  // single UTF16 code unit (U+FFFC)
+  function ContextItemsLengthUTF16(pc: pkm_core_context_item): Integer;
+  begin
+    Result := 0;
+    while pc._type <> KM_CORE_CT_END do
```

Comment:
Is it possible for context items to never contain KM_CORE_CT_END?

Reply:
No, that's contractual on the part of the Core API:

https://github.com/keymanapp/keyman/blob/41cc2f4d1e4e828c791ce8e071eb04afe5968d4d/core/include/keyman/keyman_core_api_context.h#L198-L200